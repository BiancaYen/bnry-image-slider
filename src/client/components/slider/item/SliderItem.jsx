// Slider Item

import React from 'react';
import PropTypes from 'prop-types';

// Constants
import alignmentConstants from '../../constants';

// Styles
import { ClickContainer, Container } from './styles';

// Prop Types
const propTypes = {
    children: PropTypes.node.isRequired,
    onNextClick: PropTypes.func.isRequired,
    onPreviousClick: PropTypes.func.isRequired
};

const SliderItem = ({
    children,
    onNextClick,
    onPreviousClick
}) => (
    <Container>
        <ClickContainer
            alignment={alignmentConstants.left}
            data-test="previous"
            role="button"
            tabIndex="0"
            onClick={event => onPreviousClick(event.preventDefault())}
        />
        <ClickContainer
            alignment={alignmentConstants.right}
            data-test="next"
            role="button"
            tabIndex="0"
            onClick={event => onNextClick(event.preventDefault())}
        />
        {children}
    </Container>
);

SliderItem.propTypes = propTypes;

export default SliderItem;
