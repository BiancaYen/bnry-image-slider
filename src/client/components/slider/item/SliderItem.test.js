// Slider Item Test

import { mount, shallow } from 'enzyme';
import React from 'react';
import sinon from 'sinon';

// Components
import SliderItem from './SliderItem';

// Constants
const image = (
    <img
        src="https://images.pexels.com/photos/1591447/pexels-photo-1591447.jpeg?auto=compress&cs=tinysrgb&dpr=2&w=500"
        alt="Title"
    />
);

describe('SliderItem', () => {
    it('renders with all required prop types', () => {
        const component = shallow(
            <SliderItem
                onNextClick={() => {}}
                onPreviousClick={() => {}}
            >
                {image}
            </SliderItem>
        );

        expect(component).toMatchSnapshot();
    });

    it('asserts that prop types are correct', () => {
        const onNextClick = sinon.spy();
        const onPreviousClick = sinon.spy();
        const component = mount(
            <SliderItem
                onNextClick={onNextClick}
                onPreviousClick={onPreviousClick}
            >
                {image}
            </SliderItem>
        );

        const divNext = 'div[role="button"][data-test="next"]';
        const divPrevious = 'div[role="button"][data-test="previous"]';
        expect(component.find(divNext).exists()).toEqual(true);
        expect(component.find(divPrevious).exists()).toEqual(true);
        expect(component.contains(image)).toEqual(true);

        component.find(divNext).simulate('click');
        component.find(divPrevious).simulate('click');
        expect(onNextClick).toHaveProperty('callCount', 1);
        expect(onPreviousClick).toHaveProperty('callCount', 1);
    });
});
