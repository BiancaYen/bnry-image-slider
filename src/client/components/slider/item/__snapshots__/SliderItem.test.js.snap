// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SliderItem renders with all required prop types 1`] = `
<Container>
  <ClickContainer
    alignment="left"
    data-test="previous"
    onClick={[Function]}
    role="button"
    tabIndex="0"
  />
  <ClickContainer
    alignment="right"
    data-test="next"
    onClick={[Function]}
    role="button"
    tabIndex="0"
  />
  <img
    alt="Title"
    src="https://images.pexels.com/photos/1591447/pexels-photo-1591447.jpeg?auto=compress&cs=tinysrgb&dpr=2&w=500"
  />
</Container>
`;
